File: TLE Eliminator/Level 800/halloumi_boxes.py

# Solved 
test_cases = int(input())
for i in range (test_cases):
    n,k = map(int,input().split())
    values = list(map(int,input().split()))
    flag = 1
    if k==1:
        for j in range(n-1):
            if(values[j]<=values[j+1]):
                continue
            else:
                flag = 0
                print("NO")
                break
        if(flag ==1):
            print("YES")
    else:
        print("YES")
    

File: 800 Level/halloumi_boxes.cpp
# include <bits/stdc++.h>
using namespace std;
// solved
int main(){
    int test;
    cin>>test;
    while(test--){
        int size,k;
        cin >>size>>k;
        vector <int> v(size);
        for (int i =0; i<size ; i++){
            cin>>v[i];    
        }
        bool flag = true;
        if(k==1)
        {
            for (int i =0; i<size-1 ; i++){
                if(v[i]>v[i+1])
                {
                    flag= false;
                    cout<<"NO\n";
                    break;
                }
            }
            if(flag)
            {
                 cout<<"YES\n";
            }
        }
        else{
                cout<<"YES\n";   
        } 
    }
}